EvidenceApi/V1/UseCase/UpdateDocumentSubmissionStateUseCase.cs

```diff
@@ -52,9 +59,33 @@ public DocumentSubmissionResponse Execute(Guid id, DocumentSubmissionUpdateReque
 
             documentSubmission.State = state;
 
-            if (!String.IsNullOrEmpty(request.RejectionReason))
+            if (!String.IsNullOrEmpty(request.RejectionReason) && documentSubmission.State == SubmissionState.Rejected)
             {
                 documentSubmission.RejectionReason = request.RejectionReason;
+                var evidenceRequest = _evidenceGateway.FindEvidenceRequest(documentSubmission.EvidenceRequestId);
```

Comment:
This method is becoming quite large. We could move the reject and staff selected document logic to their own methods